8.0/docker-entrypoint.sh

```diff
@@ -64,6 +64,36 @@ _get_config() {
 	# match "datadir      /some/path with/spaces in/it here" but not "--xyz=abc\n     datadir (xyz)"
 }
 
+_stop_server(){
+	# This isn't ideal, but we can generally assume only a single mysqld process is running
+	echo "Shutting down temporary server process"
+	server_pid="$1"
+	kill "$server_pid"
+	for i in $(seq 1 60); do
+		sleep 1
+		if ! $(pidof /usr/sbin/mysqld >/dev/null 2>&1); then
+			echo "Server stopped"
+			return 0
+		fi
+	done
+	# The server hasn't shut down in a timely manner
+	echo "Error: Unable to shut down server with process id $server_pid" >&2
+	return 1
+}
+_upgrade_data() {
+	local tmpdir=$1
+	chown mysql:mysql "$tmpdir"
+	# We use skip-grant-tables since we can't know the root credentials of existing databases
+	mysqld --skip-networking --pid-file="$tmpdir/mysqld.pid" --socket="$tmpdir/mysqld.sock" --daemonize --skip-grant-tables --log-error="$tmpdir/error.log"
```

Comment:
This needs to use the same `"$@"` logic we use for the other place we start the daemon, doesn't it? (which is slightly complicated by it being a shell function and `"$@"` from the script itself being obscured)

Maybe we need to _fully_ resolve what #402 starts (by functionalizing more parts of this file) so we can invoke the daemon in the same way (with a way to add `--skip-grant-tables` for this use case)?

Reply:
Yeah, I agree. I'll do some work on that first